### PR TITLE
ARROW-13303: [JS] Revise bundles

### PR DESCRIPTION
**Pull request by domoritz**

### CODE REVIEW

File: js/gulp/package-task.js [hunk 1]
```diff
@@ -46,14 +46,19 @@ const createMainPackageJson = (target, format) => (orig) => ({
     ...createTypeScriptPackageJson(target, format)(orig),
     bin: orig.bin,
     name: npmPkgName,
-    main: `${mainExport}.node`,
-    browser: `${mainExport}.dom`,
-    module: `${mainExport}.dom.mjs`,
+    type: 'commonjs',
+    main: `${mainExport}.node.js`,
```
Comment (domoritz) [inline js/gulp/package-task.js:50]:
```suggestion
    main: `${mainExport}.node`,
```

File: js/gulp/package-task.js [hunk 2]
```diff
@@ -46,14 +46,19 @@ const createMainPackageJson = (target, format) => (orig) => ({
...
+    module: `${mainExport}.dom.js`,
```
Comment (domoritz) [inline js/gulp/package-task.js:51]:
```suggestion
    module: `${mainExport}.node.js`,
```

To be consistent with https://github.com/apache/arrow/pull/10698/files#diff-9ad0361c0b3692cb897f0641d786ade504a91c8c1d6a36c3abd3cf33224946a0R96.

File: js/gulp/package-task.js [hunk 3]
```diff
@@ -46,14 +46,19 @@ const createMainPackageJson = (target, format) => (orig) => ({
...
+    browser: `${mainExport}.dom.js`,
```
Comment (domoritz) [inline js/gulp/package-task.js:52]:
```suggestion
    browser: `${mainExport}.dom`,
```

This way bundlers can pick up mjs files more easily, no?

File: js/gulp/typescript-task.js [hunk 4]
```diff
@@ -61,8 +61,8 @@ function compileTypescript(out, tsconfigPath, tsconfigOverrides) {
     return Observable.forkJoin(writeSources, writeDTypes, writeJS);
 }
 
-function cjsMapFile(mapFilePath) { return mapFilePath; }
```
Comment (domoritz) [inline js/gulp/typescript-task.js:64]: Don't we need to undo these changes so we have the correct mapping files?

**Review by domoritz — COMMENTED** [review]
If we add a file extension to a browser entry, would a builder be able to pick up an mjs file?

**Review by trxcllnt — APPROVED** [review]
